### ISSUE TITLE
*require* a 4 in DA1 to use sixels

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#2257]
Work on #2252 brought up an unexpected example: XTerm (at least version 353 as packaged in Ubuntu Focal) might be built sans sixel support, but will still happily answer `XTSMGRAPHICS` as if it could do sixel. The only indication that Sixel support is not present is the absence of a 4 in the DA1 responses.

I've verified that MLterm, WezTerm, Contour, and XTerm all return a 4 when Sixel is supported (the Alacritty ayosec/graphics branch *does not*, shittily), and @j4james has verified this for numerous other terminals. I'm tempted to require a 4 in DA1 responses to consider Sixel as supported. I think this is kinda dumb, because if you answer `XTSMGRAPHICS`, you really ought be supporting Sixel IMHO, but who knows whether that actually holds up in a historical context?

So our alternatives seem to be:
* Require '4', and fail to drive Sixel on terminals that support it, but don't supply a '4' (currently only Alacritty AFAIK). Ideally we would send patches out to get a '4' on the DA1 list.
* Require '4' or `XTSMGRAPHICS` answers, and attempt to drive Sixel on terminals lacking support (but which answer `XTSMGRAPHICS` anyway, currently only XTerm AFAIK). Ideally we would send patches out to disable `XTSMGRAPHICS` when not actually supported.

My heart says go with the second option, but my head says the first.

**Comment by dankamongmen** [comment]
```
[schwarzgerat](0) $ echo -e '\e[c\e[?2;1;1S'
[schwarzgerat](0) $ 64;1;2;6;9;15;16;17;18;21;22;28c2;0;880;552S
```

**Comment by dankamongmen** [comment]
> Yep. Alacritty is also the only terminal I know of that doesn't indicate Sixel support in `DA1` but which does respond to `XTSMGRAPHICS`. This is assuming you're willing to accept Rxvt's `?1;2;4c` response as an acceptable indicator, which would probably require some adjustments to the initial patch I suggested (maybe just replacing `da1_cb` with `da1_attrs_cb` as a quick fix).

I'd just like to take a minute to say that I'm getting kinda tired of Alacritty's branched existence -- ayosec's PR has been out roughly forever, it seems -- and tired of its [unique erasure semantics](https://github.com/dankamongmen/notcurses/issues/2142), and tired of its refusal to add XTVERSION.

i have no problem with the rxvt solution, and expect to have that finalized this evening. thanks a whole bunch again for that report.

> I think Alacritty has the same problem as Rxvt, in that it's still identifying as a VT102, so technically it's not valid to throw 4 on the end of the `DA1` response. Not sure how happy they would be with something like Rxvt's hack.

yeah, i remember this now actually -- i was going to use '4' as the litmus originally, and asked them to add it, and they had exactly this response. i'd forgotten all about that.

btw i mailed Dickey a few hours back:

```
Grandmaster Dickey,                                                                                                                 
                                                                                                                                    
I was surprised today to learn that XTerm will reply to Pi=2                                                                        
(Sixel graphics geometry) XTSMGRAPHICS queries with a                                                                               
well-formed Pi=2, Ps=0 reply even when Sixel support is not                                                                         
present (as indicated by the absence of '4' in the DA1                                                                              
response), at least in version 353 as packaged in Ubuntu Focal                                                                      
(I didn't see anything in the Changelog about this, though it                                                                       
might have been fixed in the past ~2y, in which case I apologize                                                                    
for wasting your time).                                                                                                             
                                                                                                                                    
Here I send a DA2+DA1+XTSMGRAPHICS:                                                                                                 
                                                                                                                                    
[schwarzgerat](127) $ echo -e '\e[>c\e[c\e[?2;1;1S'                                                                                 
                                                                                                                                    
^[[>41;353;0c^[[?64;1;2;6;9;15;16;17;18;21;22;28c^[[?2;0;880;552S                                                                   
       ^               ^                               ^                                                                            
       |               |                               |                                                                            
       \- version 353  |                               \- ???                                                                       
                       \- no sixel support                                                                                          
                                                                                                                                    
Given the documentation, "If configured to support either Sixel                                                                     
Graphics or ReGIS Graphics, xterm accepts a three-parameter                                                                         
control sequence, where Pi, Pa and Pv are the item, action and                                                                      
value", it seems that a valid XTSMGRAPHICS response is                                                                              
unexpected here. Neither Sixel nor Regis (DA1 feature '3') is                                                                       
available. Admittedly, it doesn't say "if and only if", but I                                                                       
feel such a phrase almost implied by the presence of "if                                                                            
configured" at all.                                                                                                                 
                                                                                                                                    
I suggest one or both of:                                                                                                           
                                                                                                                                    
* Removing the "if configured...ReGIS Graphics", and perhaps                                                                        
  even adding a note that XTSMGRAPHICS will be answered whether                                                                     
  they are supported or not (I personally find the behavior                                                                         
  quite surprising), and/or              
                                                                                                                                            
* Always responding with Ps=3 for failure, or not responding at                                                                     
  all, when lacking the appropriate graphics support (this would                                                                    
  apply only to the geometry queries, not color registers).                                                                         
                                                                                                                                    
This came up in the following issues:                                                                                               
                                                                                                                                    
 https://github.com/dankamongmen/notcurses/issues/2257                                                                              
 https://github.com/dankamongmen/notcurses/issues/2252                                                                              
                                                                                                                                    
Have a great weekend!                                                                                                               
                                                                                                                                    
--nick                                                                                                                              
```

**Comment by dankamongmen** [comment]
confirmation from Dickey that this will be fixed in XTerm (presumably 370). yay!

**Comment by dankamongmen** [comment]
alrightie, this ought be done. i think it also fixes #2255, but i need to test that.
